spark/src/main/java/org/apache/iceberg/spark/actions/BaseRewriteDataFilesSparkAction.java

```diff
@@ -383,4 +408,31 @@ public int totalGroupCount() {
       return totalGroupCount;
     }
   }
+
+  static class EmptyStruct implements StructLike {
+
+    private static EmptyStruct instance;
+
+    static EmptyStruct instance() {
```

Comment:
For most singletons, we use `public static EmptyStruct get()` and do the initialization at class init time to avoid a race condition:

```java
  private static final EmptyStruct INSTANCE = new EmptyStruct();
  static EmptyStruct get() {
    return INSTANCE;
  }
```